tools/autograd/derivatives.yaml

```diff
@@ -576,7 +576,7 @@
   self: non_differentiable
 
 - name: kthvalue(Tensor self, int k, int dim=-1, bool keepdim=False) -> (Tensor values, Tensor indices)
-  self: index_select_backward(grad, dim, indices, self.sizes(), keepdim)
+  self: select_index_backward(grad, dim, indices, self.sizes(), keepdim)
```

Comment:
These namings are quite confusing... Should we use something else now that they go into the aten API?
Like "reduction_dim_backward()"?

Reply:
`select_index_backwards` can't be use for all reductions (like mean / sum). Is there some standard name for max(), topk(), mode(), median(), max(), mean()? All of those operations perform a reduction that selects specific values.

maybe "value_selecting_reduction_backward"?

Reply:
Changed to "value_selecting_reduction_backward".